.github/workflows/builddockerimage.yml

```diff
@@ -0,0 +1,79 @@
+name: builddockerimage
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        required: true
+        type: string
+      goversion:
+        required: true
+        type: string
+      goargs:
+        required: true
+        type: string
+      labels:
+        required: true
+        type: string
+      version:
+        required: true
+        type: string
+
+jobs:
+  build_job:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: amd64
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Set up Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ inputs.goversion }}
+      - name: Setup cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            ./third_party_licenses
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-build-
+      - name: Prepare
+        run: |
+          # Get licenses if no cache exists
+          if ! [ -d ./third_party_licenses ]; then
+            go get github.com/google/go-licenses && go-licenses save ./... --save_path third_party_licenses --force
+          fi
+
+          # Get dependencies
+          go get -d ./...
+          ls -la $HOME/go/pkg/mod
+          cp -r $HOME/go/pkg/mod .
+      - name: Build target
+        uses: docker/build-push-action@v2
```

Comment:
could our multi-platform build be improved by passing all platforms directly to this action? [Documentation](https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md)

Reply:
Yes, it could. I performed a test but such a change requires more time.